pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -458,6 +461,19 @@ public static void validateIngestionAggregation(String name) {
         String.format("aggregation function %s must be one of %s", name, SUPPORTED_INGESTION_AGGREGATIONS));
   }
 
+  @VisibleForTesting
+  static void validateDecoder(StreamConfig streamConfig) {
+    if (streamConfig.getDecoderClass().equals("org.apache.pinot.plugin.inputformat.protobuf.ProtoBufMessageDecoder")) {
+      // check the existence of the needed decoder props
+      if (!streamConfig.getDecoderProperties().containsKey("stream.kafka.decoder.prop.descriptorFile")) {
```

Comment:
nit: could we put these constants to a class and have some comments to explain that these are the required fields for protobuf message decoder?

Reply:
ideally we can reference the class, but it's not on the dependency of this project.
I think the best solution to refactor this validation framework to make it pluggable, so we can move this check to the protobuf package.